lms/djangoapps/course_api/permissions.py

```diff
@@ -0,0 +1,36 @@
+"""
+Course API Authorization functions
+"""
+
+from student.roles import GlobalStaff
+
+
+def can_view_courses_for_username(requesting_user, target_username):
+    """
+    Determine whether `requesting_user` has permission to view courses available
+    to the user identified by `target_username`.  All users can view courses
+    available to themselves.  Staff users can view courses available to any given
+    user.
+
+    Arguments:
+        requesting_user (User): The user requesting permission to view another
+
+        target_username (string):
+            The name of the user `requesting_user` would like
+            to access.
+
+    Return value:
+        Boolean:
+            `True` if `requesting_user` is authorized to view courses as
+            `target_username`.  Otherwise, `False`
+    """
+    if not target_username:
+        # target_username cannot be None
+        return False
```

Comment:
If the "target_username cannot be None", shouldn't this be a coding error and therefore and assertion instead of returning a permission value of `False`.